sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceUtils.java

```diff
@@ -204,139 +204,61 @@ public static Resource getExpectedRanges(
         return resBuilder.build();
     }
 
-    public static Resource setDynamicPortName(Resource resource, String name) {
-        Labels labels = setDynamicPortName(resource.getReservation().getLabels(), name);
-
-        return Resource.newBuilder(resource)
-                .setReservation(
-                        ReservationInfo.newBuilder(resource.getReservation())
-                        .clearLabels()
-                        .setLabels(labels))
-                .build();
-    }
-
-    private static Labels setDynamicPortName(Labels labels, String name) {
-        Labels.Builder labelBuilder = Labels.newBuilder(labels);
-        labelBuilder.addLabelsBuilder()
-                .setKey(MesosResource.DYNAMIC_PORT_KEY)
-                .setValue(name);
-        return labelBuilder.build();
-    }
-
-    public static Resource setVIPPortName(Resource resource, String key, String name) {
-        Labels labels = setVIPPortName(resource.getReservation().getLabels(), key, name);
-
-        return Resource.newBuilder(resource)
-                .setReservation(
-                        ReservationInfo.newBuilder(resource.getReservation())
-                                .clearLabels()
-                                .setLabels(labels))
-                .build();
-    }
-
-    private static Labels setVIPPortName(Labels labels, String key, String name) {
-        Labels.Builder labelsBuilder = Labels.newBuilder(labels);
-        labelsBuilder.addLabels(
-                        Label.newBuilder()
-                            .setKey(MesosResource.VIP_LABEL_NAME_KEY)
-                            .setValue(key))
-                .addLabels(
-                        Label.newBuilder()
-                            .setKey(MesosResource.VIP_LABEL_VALUE_KEY)
-                            .setValue(name))
-                .build();
+    public static TaskInfo.Builder addVIP(
+            TaskInfo.Builder builder, String vipName, Integer vipPort, Resource resource) {
+        if (builder.hasDiscovery()) {
+            addVIP(builder.getDiscoveryBuilder(), vipName, vipPort, (int) resource.getRanges().getRange(0).getBegin());
+        } else {
+            builder.setDiscovery(getVIPDiscoveryInfo(builder.getName(), vipName, vipPort, resource));
+        }
 
-        return labelsBuilder.build();
+        return builder;
     }
 
-    /**
-     * Populates {@link DiscoveryInfo} with necessary information to create VIP if the provided {@link TaskInfo}
-     * has a port associated with a named VIP.
-     */
-    public static TaskInfo.Builder setDiscoveryInfo(TaskInfo.Builder taskBuilder, Collection<Resource> resources) {
-        for (Resource r : resources) {
-            if (RequirementUtils.isNamedVIPPort(r)) {
-                taskBuilder.setDiscovery(getVIPDiscoveryInfo(r, taskBuilder.getName()));
-            }
+    public static ExecutorInfo.Builder addVIP(
```

Comment:
Makes sense, if it's not actually being used there isn't much reason to add effectively-dead logic to deal with it before we've found out whether it makes to have in the first place. Maybe just add a TODO to EndpointsResource's current TaskInfo extraction for later reference?